impl/src/main/java/io/opencensus/impl/internal/DisruptorEventQueue.java

```diff
@@ -171,12 +175,17 @@ public DisruptorEvent newInstance() {
    * Every event that gets added to {@link EventQueue} will get processed here. Just calls the
    * underlying process() method.
    */
-  private static final class DisruptorEventHandler implements EventHandler<DisruptorEvent> {
+  private enum DisruptorEventHandler implements EventHandler<DisruptorEvent> {
+    INSTANCE;
+
     @Override
-    // TODO(sebright): Fix the Checker Framework warning.
-    @SuppressWarnings("nullness")
     public void onEvent(DisruptorEvent event, long sequence, boolean endOfBatch) {
-      event.getEntry().process();
+      Entry entry = event.getEntry();
+      if (entry != null) {
+        entry.process();
+      }
+      // Remove the reference to the previous entry to allow the memory to be gc'ed.
+      event.setEntry(null);
```

Comment:
Just curious: `DisruptorEvent` will never be gc'ed?

Reply:
That is correct and an intended behavior. `DisruptorEvent` is a small object and in total we hold 8KB memory here which should be fine.